### PR TITLE
[DISCO-2961] Add categories to Nav Suggest

### PR DESCRIPTION
**Pull request by tiftran**
## References

JIRA: [DISCO-2961](https://mozilla-hub.atlassian.net/browse/DISCO-2961)

## Description
Added categories to Nav Suggestions.

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [ ] This PR conforms to the [Contribution Guidelines](https://github.com/mozilla-services/merino-py/blob/main/CONTRIBUTING.md)
- [ ] The PR title starts with the JIRA issue reference, format `[DISCO-####]`, and has the same title (if applicable)
- [ ] `[load test: (abort|warn)]` keywords are applied (if applicable)
- [ ] [Documentation](https://github.com/mozilla-services/merino-py/tree/main/docs) has been updated (if applicable)
- [ ] [Functional and performance test](https://github.com/mozilla-services/merino-py/blob/main/docs/dev/testing.md) coverage has been expanded and maintained (if applicable)

[DISCO-2961]: https://mozilla-hub.atlassian.net/browse/DISCO-2961?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Herraj — APPROVED** [review]
🚀

File: merino/jobs/navigational_suggestions/__init__.py [hunk 1]
```diff
@@ -84,19 +89,33 @@ def _construct_top_picks(
...
+                    "serp_categories": _get_serp_categories(domain_url),
+                    "url": domain_url,
                     "title": domain_metadata[index]["title"],
                     "icon": favicons[index],
                 }
             )
     return {"domains": result}
 
 
+def _get_serp_categories(domain_url: str | None):
```
Comment (ncloudioj) [inline merino/jobs/navigational_suggestions/__init__.py:107]: Let's annotate the return type.

**Review by ncloudioj — APPROVED** [review]
Looks great, thanks!

Just wanted to quickly confirm that we use the same threshold as R2D2 to filter domains in domain_category_mapping.py, correct?

**Comment by tiftran** [comment]
@ncloudioj yup, I believe i filtered out with the same threshold value. Did a couple spot checks just to be sure

**Pull request merged**